### PR TITLE
Update README with dependency build instructions

### PR DESCRIPTION
**Pull request by romanroibu**

### CODE REVIEW

**Comment by romanroibu** [comment]
On macOS, `make` fails with the following
```
[ 99%] Linking CXX shared library ../../lib/libopencv_world.dylib
Undefined symbols for architecture x86_64:
  "_jpeg_default_qtables", referenced from:
      cv::JpegEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in grfmt_jpeg.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libopencv_world.4.2.0.dylib] Error 1
make[1]: *** [modules/world/CMakeFiles/opencv_world.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by romanroibu** [comment]
On Ubuntu the dependencies build successfully

**Comment by romanroibu** [comment]
On Windows the dependencies build successfully

File: README.md [hunk 1]
```diff
@@ -1,3 +1,199 @@
...
+
+### Build Dependencies
+
+#### Windows
+
+##### Build and install OpenCV
+
+```bash
+# Download OpenCV
+Invoke-WebRequest "https://github.com/opencv/opencv/archive/4.2.0.zip" -OutFile opencv.zip
```
Comment (papr) [inline README.md:17]: Please add a comment that this is a `powershell` command

File: README.md [hunk 2]
```diff
@@ -1,3 +1,199 @@
...
+
+# Exit build path
+cd ../..
+```
+
+##### Build and install Eigen
+
+```bash
+# Download Eigen
+Invoke-WebRequest "https://gitlab.com/libeigen/eigen/-/archive/3.3.7/eigen-3.3.7.zip" -OutFile eigen.zip
```
Comment (papr) [inline README.md:41]: Same here please

File: README.md [hunk 3]
```diff
@@ -1,3 +1,199 @@
...
+cd ../..
+```
+
+#### macOS
+
+##### Build and install OpenCV
+
+```bash
+# Download OpenCV
+wget -q -O opencv.zip "https://github.com/opencv/opencv/archive/4.2.0.zip"
```
Comment (papr) [inline README.md:137]: wget is not installed by default on macOS AFAIK, maybe we add a comment on how to install it via brew?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,199 @@
...
+
+# Exit build path
+cd ../..
+```
+
+##### Build and install Eigen
+
+```bash
+# Download Eigen
+wget -q -O eigen.zip "https://gitlab.com/libeigen/eigen/-/archive/3.3.7/eigen-3.3.7.zip"
```
Comment (papr) [inline README.md:179]: Same here

**Review by papr — COMMENTED** [review]
What is your cmake config output on macOS? Maybe we can check what JPEG lib it uses.

**Comment by papr** [comment]
Please also remove the `-q` flags to give users feedback regarding progress.

**Comment by papr** [comment]
My cmake config says:
> JPEG:                        build-libjpeg-turbo (ver 2.0.2-62)

@romanroibu what does yours?

**Comment by romanroibu** [comment]
@papr 
> JPEG: build-libjpeg-turbo (ver 2.0.2-62)

<details>
<summary>Full cmake config output</summary>

```bash
-- Detected processor: x86_64
-- Looking for ccache - not found
-- AVX512_KNM is not supported by C++ compiler
-- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES) 
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND) 
-- libjpeg-turbo: VERSION = 2.0.2, BUILD = opencv-4.2.0-libjpeg-turbo
-- found Intel IPP (ICV version): 2019.0.0 [2019.0.0 Gold]
-- at: /Users/romanroibu/work/pye3d-detector/opencv/build/3rdparty/ippicv/ippicv_mac/icv
-- found Intel IPP Integration Wrappers sources: 2019.0.0
-- at: /Users/romanroibu/work/pye3d-detector/opencv/build/3rdparty/ippicv/ippicv_mac/iw
-- LAPACK(LAPACK/Apple): LAPACK_LIBRARIES: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/System/Library/Frameworks/Accelerate.framework;-lm;-ldl
-- LAPACK(LAPACK/Apple): Support is enabled.
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.

-- Could NOT find JNI (missing: JAVA_INCLUDE_PATH JAVA_INCLUDE_PATH2 JAVA_AWT_INCLUDE_PATH) 
-- VTK is not found. Please set -DVTK_DIR in CMake to VTK build directory, or to VTK install subdirectory with VTKConfig.cmake file
-- OpenCV Python: during development append to PYTHONPATH: /Users/romanroibu/work/pye3d-detector/opencv/build/python_loader
-- Checking for module 'gstreamer-base-1.0'
--   No package 'gstreamer-base-1.0' found
-- Checking for module 'gstreamer-app-1.0'
--   No package 'gstreamer-app-1.0' found
-- Checking for module 'gstreamer-riff-1.0'
--   No package 'gstreamer-riff-1.0' found
-- Checking for module 'gstreamer-pbutils-1.0'
--   No package 'gstreamer-pbutils-1.0' found
-- Checking for module 'libdc1394-2'
--   No package 'libdc1394-2' found
-- Using whitelist: opencv_core;opencv_highgui;opencv_imgcodecs;opencv_imgproc;opencv_video;opencv_videoio;opencv_world
-- Module opencv_calib3d disabled by whitelist
-- Module opencv_dnn disabled by whitelist
-- Module opencv_features2d disabled by whitelist
-- Module opencv_flann disabled by whitelist
-- Module opencv_gapi disabled by whitelist
-- Module opencv_java_bindings_generator disabled by whitelist
-- Module opencv_ml disabled by whitelist
-- Module opencv_objdetect disabled by whitelist
-- Module opencv_photo disabled by whitelist
-- Module opencv_python_bindings_generator disabled by whitelist
-- Module opencv_python_tests disabled by whitelist
-- Module opencv_stitching disabled by whitelist
-- Processing WORLD modules...
--     module opencv_core...
--     module opencv_imgproc...
--     module opencv_video...
--     module opencv_imgcodecs...
--     module opencv_videoio...
--     module opencv_highgui...
-- Processing WORLD modules... DONE
-- 
-- General configuration for OpenCV 4.2.0 =====================================
--   Version control:               4ad96e7
-- 
--   Platform:
--     Timestamp:                   2020-12-02T19:16:12Z
--     Host:                        Darwin 20.1.0 x86_64
--     CMake:                       3.18.4
--     CMake generator:             Unix Makefiles
--     CMake build tool:            /usr/bin/make
--     Configuration:               Release
-- 
--   CPU/HW features:
--     Baseline:                    SSE SSE2 SSE3 SSSE3 SSE4_1
--       requested:                 DETECT
--     Dispatched code generation:  SSE4_2 FP16 AVX AVX2 AVX512_SKX
--       requested:                 SSE4_1 SSE4_2 AVX FP16 AVX2 AVX512_SKX
--       SSE4_2 (1 files):          + POPCNT SSE4_2
--       FP16 (0 files):            + POPCNT SSE4_2 FP16 AVX
--       AVX (3 files):             + POPCNT SSE4_2 AVX
--       AVX2 (23 files):           + POPCNT SSE4_2 FP16 FMA3 AVX AVX2
--       AVX512_SKX (2 files):      + POPCNT SSE4_2 FP16 FMA3 AVX AVX2 AVX_512F AVX512_COMMON AVX512_SKX
-- 
--   C/C++:
--     Built as dynamic libs?:      YES
--     C++ Compiler:                /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  (ver 12.0.0.12000032)
--     C++ flags (Release):         -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG
--     C++ flags (Debug):           -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     C Compiler:                  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
--     C flags (Release):           -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG
--     C flags (Debug):             -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Wno-long-long -Qunused-arguments -Wno-semicolon-before-method-body -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     Linker flags (Release):      -Wl,-dead_strip  
--     Linker flags (Debug):        -Wl,-dead_strip  
--     ccache:                      NO
--     Precompiled headers:         NO
--     Extra dependencies:
--     3rdparty dependencies:
-- 
--   OpenCV modules:
--     To be built:                 core highgui imgcodecs imgproc video videoio world
--     Disabled:                    -
--     Disabled by dependency:      calib3d dnn features2d flann gapi java_bindings_generator ml objdetect photo python_bindings_generator python_tests stitching
--     Unavailable:                 java js python2 python3 ts
--     Applications:                apps
--     Documentation:               NO
--     Non-free algorithms:         NO
-- 
--   GUI: 
--     Cocoa:                       YES
--     VTK support:                 NO
-- 
--   Media I/O: 
--     ZLib:                        zlib (ver 1.2.11)
--     JPEG:                        build-libjpeg-turbo (ver 2.0.2-62)
--     WEBP:                        build (ver encoder: 0x020e)
--     PNG:                         libpng (ver 1.6.37)
--     TIFF:                        build (ver 42 - 4.0.10)
--     JPEG 2000:                   build (ver 1.900.1)
--     OpenEXR:                     build (ver 2.3.0)
--     HDR:                         YES
--     SUNRASTER:                   YES
--     PXM:                         YES
--     PFM:                         YES
-- 
--   Video I/O:
--     DC1394:                      NO
--     FFMPEG:                      YES
--       avcodec:                   YES (58.91.100)
--       avformat:                  YES (58.45.100)
--       avutil:                    YES (56.51.100)
--       swscale:                   YES (5.7.100)
--       avresample:                YES (4.0.0)
--     GStreamer:                   NO
--     AVFoundation:                YES
-- 
--   Parallel framework:            GCD
-- 
--   Trace:                         YES (with Intel ITT)
-- 
--   Other third-party libraries:
--     Intel IPP:                   2019.0.0 Gold [2019.0.0]
--            at:                   /Users/romanroibu/work/pye3d-detector/opencv/build/3rdparty/ippicv/ippicv_mac/icv
--     Intel IPP IW:                sources (2019.0.0)
--               at:                /Users/romanroibu/work/pye3d-detector/opencv/build/3rdparty/ippicv/ippicv_mac/iw
--     Lapack:                      YES (/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/System/Library/Frameworks/Accelerate.framework -lm -ldl)
--     Eigen:                       YES (ver 3.3.8)
--     Custom HAL:                  NO
--     Protobuf:                    build (3.5.1)
-- 
--   OpenCL:                        YES (no extra features)
--     Include path:                NO
--     Link libraries:              -framework OpenCL
-- 
--   Python (for build):            /usr/bin/python2.7
-- 
--   Java:                          
--     ant:                         NO
--     JNI:                         NO
--     Java wrappers:               NO
--     Java tests:                  NO
-- 
--   Install to:                    /Users/romanroibu/work/pye3d-detector/opencv/build
-- -----------------------------------------------------------------
-- 
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/romanroibu/work/pye3d-detector/opencv/build
```
</details>

Does it build successfully on your machine?

**Comment by papr** [comment]
@romanroibu It only does if I disable OpenMP because it cannot find `omp.h` on my computer.

**Review by papr — APPROVED** [review]

**Pull request merged**